src/scene/mesh/shared/MeshPipe.ts

```diff
@@ -89,11 +91,24 @@ export class MeshPipe implements RenderPipe<Mesh>, InstructionPipe<MeshInstructi
 
         if (wasBatched !== isBatched)
         {
+            if (isBatched)
+            {
+                const geometry = mesh._geometry as MeshGeometry;
+
+                meshData.indexSize = geometry.indices.length;
+                meshData.vertexSize = geometry.positions.length;
+            }
+            else
+            {
+                meshData.indexSize = 0;
+                meshData.vertexSize = 0;
+            }
+
```

Comment:
I just noticed this in passing: if the `indexSize` and `vertexSize` is not updated in this branch, I expect a second rebuild if the shader from a mesh is removed and at the same time the geometry is changed. Because in the next branch the `indexSize`/`vertexSize` are not equal.